### PR TITLE
Feature lifecycle split api core

### PR DESCRIPTION
**Pull request by Verbalist**
https://trello.com/c/ufNMujr8/126-check-transaction-lifecycle

### CODE REVIEW

File: config/default/genesisBlock.json [hunk 1]
```diff
@@ -1 +1,103 @@
...
+  "transactionCount": 7,
+  "amount": 4797000000000000,
+  "fee": 482700000000,
+  "payloadHash": "7e6ba6ec459d96207414f61b67ecd2ecd8c946507bb102a1e47a0ce987e494d0",
+  "generatorPublicKey": "f4ae589b02f97e9ab5bce61cf187bcc96cfb3fdf9a11333703a682b7d47c8dc2",
+  "signature": "d664da064f083be23d8845c3dc2572342dfbb3a04c037205d3fee8a973dd7a73dfb1e6dafcdb06b9738c9d7be4f0e5e98f237187f055edb8c307d6cbfa457207",
+  "transactions": [
+    {
+      "id": "2c52682e6a51a9ddfd48a679a95c9fea4e693790aec5968535a482088b6c75bf",
+      "block_id": "cbb9449abb9672d33fa2eb200b1c8b03db7c6572dfb6e59dc334c0ab82b63ab0",
```
Comment (axelsheva) [inline config/default/genesisBlock.json]: here must be camel case

File: core/controller/transaction.ts [hunk 2]
```diff
@@ -1,28 +1,26 @@
...
-import AccountRepository from 'core/repository/account';
 
 class TransactionController extends BaseController {
     @ON('TRANSACTION_RECEIVE')
     public async onReceiveTransaction(action: { data: { trs: TransactionModel<IAsset> } }): Promise<void> {
         const { data } = action;
         logger.debug(`[Controller][Transaction][onReceiveTransaction] ${JSON.stringify(data.trs)}`);
 
-        if (!TransactionService.validate(data.trs, AccountRepository.getByPublicKey(data.trs.senderPublicKey))) {
+        if (TransactionPool.has(data.trs)) {
```
Comment (axelsheva) [inline core/controller/transaction.ts]: I think we need to validate transaction first

File: core/service/transaction/stake.ts [hunk 3]
```diff
@@ -87,15 +86,16 @@ class TransactionStakeService implements IAssetService<IAssetStake>  {
         return (trs.asset.amount * config.constants.fees.froze) / TOTAL_PERCENTAGE;
     }
 
-    applyUnconfirmed(trs: Transaction<IAssetStake>): void {
-        const totalAmount: number = trs.fee + trs.asset.amount;
-        AccountRepo.updateBalanceByAddress(trs.senderAddress, -totalAmount);
+    applyUnconfirmed(trs: Transaction<IAssetStake>, sender: Account): void {
+        sender.actualBalance -= trs.asset.amount;
+        // TODO sender.stakes.push(new Stake())
+        // distribute airdrop
```
Comment (axelsheva) [inline core/service/transaction/stake.ts]: for a stake, we will send direct reward?

File: core/controller/transaction.ts (hunk 2 above)
Comment (Verbalist) [reply core/controller/transaction.ts]: validate more expensive operation then has. If id doesn't exist has will work correct

File: core/service/transaction/stake.ts (hunk 3 above)
Comment (Verbalist) [reply core/service/transaction/stake.ts]: Yes, we will change airdrop and recipient account for airdrop here

File: config/default/genesisBlock.json (hunk 1 above)
Comment (Verbalist) [reply config/default/genesisBlock.json]: Not me. Say it to @alex-arugon

File: core/controller/transaction.ts (hunk 2 above)
Comment (axelsheva) [reply core/controller/transaction.ts]: but when trs will be `undefined`?

**Pull request merged**